src/core/notification.rs

```diff
@@ -0,0 +1,191 @@
+use std::sync::atomic::AtomicU8;
+use std::sync::OnceLock;
+use std::{
+    sync::{mpsc, Arc, Mutex},
+    time::{Duration, Instant},
+};
+
+use imgui::{Ui, WindowFlags};
+use log::Level;
+
+#[derive(Debug, Clone, Copy, PartialEq, Default)]
+enum State {
+    #[default]
+    Normal,
+    Fading{ num: f32, out: bool },
+    Dismissed,
+}
+
+#[derive(Debug, Clone)]
+struct Message {
+    body: String,
+    level: Level,
+    created_at: Instant,
+    state: State,
+    id: u8,
+}
+
+const WIN_SIZE: [f32; 2] = [160f32, -1.0];
+const Y_PADDING: f32 = 12.0;
+const X_PADDING: f32 = Y_PADDING;
+const TIMEOUT: Duration = Duration::from_secs(5);
+
+impl Message {
+    fn draw(&mut self, ui: &Ui, x: f32, y: f32) -> Option<f32> {
+        let winflags = WindowFlags::NO_RESIZE
+            | WindowFlags::NO_MOVE
+            /* | WindowFlags::NO_FOCUS_ON_APPEARING
+            | WindowFlags::NO_BRING_TO_FRONT_ON_FOCUS */
+            | WindowFlags::NO_SAVED_SETTINGS
+            | WindowFlags::NO_COLLAPSE;
+        let tk = ui.push_style_color(imgui::StyleColor::Text, level_color(self.level));
+        let alpha = if let State::Fading{ num, .. } = self.state {
+            Some(ui.push_style_var(imgui::StyleVar::Alpha(num)))
+        } else { None };
+        let (pressed, y) = ui
+            .window(format!("{} {}##{}", level_icon(self.level), self.level, self.id))
+            .flags(winflags)
+            .position([x, y], imgui::Condition::Always)
+            .size(WIN_SIZE, imgui::Condition::Always)
+            .build(|| {
+                tk.pop();
+                ui.text(self.body.as_str());
+                // Start dismissing the notification
+                let pressed = ui.button("Ok");
+                return (pressed, ui.window_size()[1] + Y_PADDING);
```

Comment:
```suggestion
                (pressed, ui.window_size()[1] + Y_PADDING)
```
Imagino que o `return` sobrou como resquício de refatoração

Reply:
Por uma questão de estilo, eu costumo não fazer retornos implícitos em funções mais procedurais. Eu uso mais quando o propósito principal é computar valores ou se a função for pequena.

Não me importaria de aplicar as mudanças sugeridas, porém